cardano-client/src/Cardano/Client/API2.hs

```diff
@@ -0,0 +1,318 @@
+    , NodeToClientVersionData (..)
+--    , SubscriptionParams (..)
+    , WithAddr (..)
+    , ncSubscriptionWorker_V1
+    , networkErrorPolicies
+    , newNetworkMutableState
+    , nodeToClientCodecCBORTerm
+-- Ouroboros.Network.Protocol.ChainSync.ClientPipelined
+--    , ChainSyncClientPipelined (..)
+--  , ClientPipelinedStIdle (..)
+--   , ClientPipelinedStIdle  ( SendMsgFindIntersect
+--                           , SendMsgRequestNext
+--                           , SendMsgRequestNextPipelined
+--                           , CollectResponse)
+    , ClientStNext (..)
+--    , chainSyncClientPeerPipelined -- cardano-explorer
+--    , ClientPipelinedStIntersect (..)
+-- Ouroboros.Network.Protocol.ChainSync.PipelineDecision
+    , pipelineDecisionLowHighMark
+    , PipelineDecision (..)
+    , runPipelineDecision
+    , MkPipelineDecision
+-- Ouroboros.Network.Protocol.ChainSync.Codec
+    , codecChainSync
+-- Ouroboros.Network.Protocol.ChainSync.Type
+    , ChainSync
+-- Ouroboros.Network.Protocol.LocalTxSubmission.Client
+    , LocalTxSubmissionClient (..)
+    , LocalTxClientStIdle (SendMsgSubmitTx)
+    , localTxSubmissionClientPeer
+-- Ouroboros.Network.Protocol.LocalTxSubmission.Codec
+    , codecLocalTxSubmission
+-- Ouroboros.Network.Protocol.LocalTxSubmission.Type
+    , LocalTxSubmission
+-- Network.TypedProtocol.Codec
+    , Codec
+-- Network.TypedProtocol.Codec.Cbor
+--    , DeserialiseFailure
+-- Network.TypedProtocol.Driver
+    , runPeer
+    , runPipelinedPeer
+-- Network.TypedProtocol.Pipelined
+    , Nat(Zero, Succ)
+-- Ouroboros.Consensus.Ledger.Byron
+    , ByronBlock (ByronBlock, byronBlockRaw, byronBlockSlotNo, byronBlockHash)
+    , ByronHash (..)
+    , GenTx
+-- Ouroboros.Consensus.Ledger.Abstract
+    , BlockProtocol
+-- Ouroboros.Consensus.Node.ProtocolInfo
+    , pInfoConfig
+    , protocolInfo
+-- Ouroboros.Consensus.Node.Run.Abstract
+    , RunNode
+    , nodeDecodeBlock
+    , nodeDecodeGenTx
+    , nodeDecodeHeaderHash
+    , nodeEncodeBlock
+    , nodeEncodeGenTx
+    , nodeEncodeHeaderHash
+    , nodeNetworkMagic
+-- Ouroboros.Consensus.Node.ErrorPolicy
+    , consensusErrorPolicy
+-- Ouroboros.Consensus.Protocol
+    , NodeConfig
+    , Protocol (..)
+
+-- wallet extra imports
+-- Codec.SerialiseTerm
+    , CodecCBORTerm
+-- Ouroboros.Network.Magic
+    , NetworkMagic (..)
+-- import qualified Ouroboros.Network.Block as O
+-- qualified Ouroboros.Network.Point as Point
+-- Ouroboros.Network.ChainFragment
+    , HasHeader (..) -- blockNo
+
+-- import Network.TypedProtocol.Channel
+    , Channel
+-- import Network.TypedProtocol.Driver
+    , TraceSendRecv
+-- import Ouroboros.Consensus.Ledger.Byron
+--    , ByronBlock (..)
+    , decodeByronBlock
+    , decodeByronGenTx
+    , decodeByronHeaderHash
+    , encodeByronBlock
+    , encodeByronGenTx
+    , encodeByronHeaderHash
+-- import Ouroboros.Network.NodeToClient
+    , ConnectionId (..)
+    , NetworkConnectTracers (..)
+    , NodeToClientVersion (..)
```

Comment:
We need to think how to export it. It will clash with `NodeToClientVersion` in `ouroboros-consensus` which I think is more useful (but it would be good to check in various clients).